src/docfx/lib/git/CommitBuildTimeProvider.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Docs.Build
+{
+    internal class CommitBuildTimeProvider
+    {
+        private readonly Repository _repo;
+        private readonly string _commitBuildTimePath;
+        private readonly IReadOnlyDictionary<string, DateTime> _buildTimeByCommit;
+
+        public CommitBuildTimeProvider(Repository repo)
+        {
+            _repo = repo;
+            _commitBuildTimePath = AppData.GetCommitBuildTimePath(repo.Remote, repo.Branch);
+
+            var commitBuildTime = new CommitBuildTime();
+            if (File.Exists(_commitBuildTimePath))
+            {
+                Log.Write($"Using git commit build time cache file: '{_commitBuildTimePath}'");
+                commitBuildTime = JsonUtility.Deserialize<CommitBuildTime>(ProcessUtility.ReadFile(_commitBuildTimePath));
+            }
+
+            _buildTimeByCommit = commitBuildTime.Commits.ToDictionary(item => item.Sha, item => item.BuiltAt);
+        }
+
+        public bool TryGetCommitBuildTime(string commitId, out DateTime time)
+        {
+            return _buildTimeByCommit.TryGetValue(commitId, out time);
+        }
+
+        public void UpdateAndSaveCache()
```

Comment:
update test case to cover UpdateAndSave?

Reply:
that part is a bit tricky to do with spec test, since the result is unstable (built_at = now)

Reply:
use `commit` time for this test?